homeassistant/components/otp/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for One-Time Password (OTP) integration."""
+
+from __future__ import annotations
+
+import binascii
+import logging
+from typing import Any
+
+import pyotp
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_NAME, CONF_TOKEN
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_TOKEN): str,
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+    }
+)
+
+
+class TOTPConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for One-Time Password (OTP)."""
+
+    VERSION = 1
+    user_input: dict[str, Any]
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await self.hass.async_add_executor_job(
+                    pyotp.TOTP(user_input[CONF_TOKEN]).now
+                )
+            except binascii.Error:
+                errors["base"] = "invalid_code"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(user_input[CONF_TOKEN])
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
```

Comment:
Can we automatically get the name from the token?
If you scan the QR code in the authenticator app, it will automatically suggest the title

Reply:
The Token itself does not have a Name encoded, only the QR Code has.

Reply:
As you can see in this codesnippet from the first commit were I had the QR code generation, the QR code contains a provisioning_uri, which is constructed from the token, a username and an issuer name. When configuring the OTP sensor from the Token, there is no additional metadata that can be extracted from it. But it could be a really cool feature, if the Config flow would access the camera to scan such a QR code and setup the sensor from it 😎

```python
        url = await self.hass.async_add_executor_job(
            pyotp.TOTP(self.user_input[CONF_TOKEN]).provisioning_uri,
            self.user_input[CONF_NAME],
            "Home Assistant",
        )
        qr_code = await self.hass.async_add_executor_job(_generate_qr_code, url)
```